pkg/servicemesh/federation/common/util.go

```diff
@@ -0,0 +1,55 @@
+// Copyright Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	kubelabels "k8s.io/apimachinery/pkg/labels"
+
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/pkg/spiffe"
+)
+
+func EndpointsForService(client kube.Client, name, namespace string) (*corev1.Endpoints, error) {
+	return client.KubeInformer().Core().V1().Endpoints().Lister().Endpoints(namespace).Get(name)
+}
+
+func ServiceAccountsForService(client kube.Client, name, namespace string) (map[string]string, error) {
```

Comment:
It would be great to have comments for exported/public functions, e.g., so that we know what to expect as a result. For instance, on error, why not return nil instead of an empty map?